hasher-matcher-actioner/hmalib/matchers/matchers_base.py

```diff
@@ -58,6 +58,49 @@ def get_privacy_group_matcher_active(privacy_group_id: str) -> bool:
     )
 
 
+@functools.lru_cache(maxsize=128)
+def _get_all_matcher_active_privacy_groups(cache_buster) -> t.List[str]:
+    configs = ThreatExchangeConfig.get_all()
+    return list(
+        map(
+            lambda c: c.name,
+            filter(
+                lambda c: c.matcher_active,
+                configs,
+            ),
+        )
+    )
+
+
+@functools.lru_cache(maxsize=128)
+def _get_all_privacy_group_active_matcher_pdq_theshold(
+    cache_buster,
+) -> t.Dict[str, int]:
+    active_pg_names = _get_all_matcher_active_privacy_groups(cache_buster)
+    if not active_pg_names:
+        return {}
+    return {
+        config.name: config.pdq_match_threshold
+        for config in AdditionalMatchSettingsConfig.get_all()
+        if config.name in active_pg_names
+    }
+
+
+def get_max_threshold_of_active_privacy_groups_for_signal_type(
+    signal_type: t.Type[SignalType],
+) -> int:
+    if signal_type == PdqSignal:
+        pg_to_pdq_threshold = _get_all_privacy_group_active_matcher_pdq_theshold(
+            time.time() // PG_CONFIG_CACHE_TIME_SECONDS
+        )
+        # no custom threshold set/active
+        if not pg_to_pdq_threshold:
+            return 0
+        return max(pg_to_pdq_threshold.values())
```

Comment:
nit: this way, you end up computing `max` for every match. Could this be moved into the cached function?